### PR TITLE
Update delayJS script and stop handling scripts inside svg

### LINKED ISSUE #5725: Update the Delay JS script to the latest version
**Before submitting an issue please check that you’ve completed the following steps:**
- Made sure you’re on the latest version
- Used the search feature to ensure that the bug hasn’t been reported before

**Describe the bug**
On some pages, some scripts are called twice. And when they execute, they'll overwrite itselves and other instances. When the second delayed script tries to execute, it's already disappeared from the DOM and DJE fails to transform it.

This can cause compatibility problems.

**Additional context**
We need to update the script with the latest [v1.2](https://github.com/wp-media/delay-javascript-loading/pull/60) changes.

**Backlog Grooming (for WP Media dev team use only)**
- [ ] Reproduce the problem
- [ ] Identify the root cause
- [ ] Scope a solution
- [ ] Estimate the effort

### PR DESCRIPTION
**Pull request by CrochetFeve0251**
## Description
Updated lazyload JS script.

Fixes #5725

## Type of change

Please delete options that are not relevant.

- [ ] Bug

## How Has This Been Tested?

- [ ] Automated test 

# Checklist:

Please delete the options that are not relevant.

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by piotrbak** [comment]
@wp-media/php For the 3.12.5.2 we'll also need to make sure that `<script>` inside `< svg >`  is not rewritten to Delay JS.

Related convo:
https://wp-media.slack.com/archives/GUT7FLHF1/p1675189916149859

File: inc/Engine/Optimization/RegexTrait.php [hunk 1]
```diff
@@ -110,6 +118,40 @@ protected function replace_xmp_tags( $html ) {
 		return $replaced_html;
 	}
 
+	/**
+	 * Replace <xmp> tags in the HTML with comment
```
Comment (jeawhanlee) [inline inc/Engine/Optimization/RegexTrait.php]:
```suggestion
	 * Replace <svg> tags in the HTML with comment
```

File: inc/Engine/Optimization/RegexTrait.php [hunk 2]
```diff
@@ -110,6 +118,40 @@ protected function replace_xmp_tags( $html ) {
...
+	 *
+	 * @since 3.12.3
```
Comment (jeawhanlee) [inline inc/Engine/Optimization/RegexTrait.php]:
```suggestion
	 * @since 3.12.5.3
```

File: inc/Engine/Optimization/RegexTrait.php [hunk 3]
```diff
@@ -124,6 +166,22 @@ protected function replace_xmp( $match ) {
 		return $key;
 	}
 
+	/**
+	 * Restore <svg> tags
+	 *
+	 * @since 3.12.3
```
Comment (jeawhanlee) [inline inc/Engine/Optimization/RegexTrait.php]:
```suggestion
	 * @since 3.12.5.3
```

**Review by jeawhanlee — CHANGES_REQUESTED** [review]

**Review by jeawhanlee — APPROVED** [review]

**Comment by charls637** [comment]
On firefox, when delayJS script  is enabled.. links are not working A links and Nav links.. Using elementor.. btw chrome and other browser works fine.

**Comment by piotrbak** [comment]
@charls637 Hello and thanks for your comment. Are you using the latest, 3.12.6.1 version of WP Rocket?

If so, and you still experience this problem, please reach our Support.

**Comment by charls637** [comment]
image
Hello @piotrbak  thank you for the fast response. I already submitted the ticket  5-6 hours ago.
Latest I can use is wp-rocket_3.12.5.3.zip.. I tried to download it from the account page.

**Pull request merged**